File: serverless/friday-newsletter/src/lambda_handler/lambda_request.rs
use serde_derive::{Deserialize, Serialize};

#[derive(Debug, Deserialize, Serialize)]
pub struct LambdaRequest {
    pub action: String,
    pub data: serde_json::Value,
    #[serde(rename = "correlationId")]
    pub correlation_id: Option<String>
}

File: serverless/FridayStateMachine/NewsletterSagas/NewsletterStateMachine/Contracts/LambdaRequest.cs
namespace NewsletterStateMachine.Contracts;

public class LambdaRequest<TPayload>
{
    public string Action { get; set; } = string.Empty;
    public TPayload? Data { get; set; }
    public string CorrelationId { get; set; } = string.Empty;
}
